### PR TITLE
[PoC] My term trait 

### PR DESCRIPTION
**Pull request by MattesWhite**
This PR implements my proposal from [my comment](https://github.com/pchampin/sophia_rs/pull/69#commitcomment-38553899) in #69.

This version adds no further types expect for the traits. The traits are implemented for the respective, existing types and an example function is given with `Graph::poc_triples_with_spo()`.

---

The contents are a _Proof of Concept_ and neither completely thought through nor completely implemented. Rather it is a base for further discussion in #69.

### CODE REVIEW

File: term/src/traits.rs [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    fn as_literal(&self) -> Option<&dyn Literal> {
+        None
+    }
+    /// Downcast to a blank node.
+    fn as_blank_node(&self) -> Option<&dyn BlankNode> {
+        None
+    }
+}
+
+impl<'a> PartialEq for &'a dyn Term {
```
Comment (pchampin) [inline term/src/traits.rs:25]: Darn, I always forget we can do this (implementing a trait for a *reference* type). Nice.

**Comment by MattesWhite** [comment]
Sorry for the merge misunderstood something... well this isn't going to be merged anyway...

**Comment by MattesWhite** [comment]
In the latest commit I introduce `IriView` which prevents `traits::Term` from returning a trait object. But `IriView` is actually only a wrapper for `&dyn Iri` so nothing won with that solution. However, I think this could be a valuable direction to go. I try further alternatives.

**Comment by MattesWhite** [comment]
As @pchampin is working on an introduction of a `Term` trait himself (which we agreed on in another online discussion), I close this PR.

**Pull request closed without merging**